modules/ding_ddbasic/modules/ding_ddbasic_opening_hours/ding_ddbasic_opening_hours.theme.inc

```diff
@@ -60,9 +60,11 @@ function template_preprocess_ding_ddbasic_opening_hours_week(&$variables) {
   }
 
   usort($instances, 'ding_ddbasic_opening_hours_sort');
-  // Add true NULL to the categories.
-  if (in_array('_null', $variables['categories'])) {
-    $variables['categories'][NULL] = NULL;
+
+  // Add NULL to the categories, if the the NO_CATEGORY category is selected.
+  // So an empty category_tid of the instance matches the selected categories.
+  if (in_array(DING_DDBASIC_OPENING_HOURS_NO_CATEGORY, $variables['categories'])) {
+    $variables['categories'][''] = NULL;
```

Comment:
The opening_hours module defaults it's category_tid to NULL, so we need a NULL value in the $variables['categories'] to "catch" this category, while adding the instances to the structured dataset (see the first if statement in the foreach below).
Note we only add the NULL if the user has selected to filter on the default category (NO_CATEGORY).